io1bit/io1bit_magma.py

```diff
@@ -0,0 +1,68 @@
+import magma
+from generator.configurable import ConfigurationType
+from common.core import Core
+from generator.from_magma import FromMagma
+from io1bit import tristate_genesis2
+
+
+class IO1bit(Core):
+    def __init__(self, io_group, side):
+        super().__init__()
+
+        wrapper = tristate_genesis2.tristate_wrapper
+        generator = wrapper.generator(mode="declare")
+        tristate_circ = generator()
+        self.tristate = FromMagma(tristate_circ)
+
+        # TODO: I think we can remove these two parameters
+        self.io_group = io_group
+        self.side = side
+        TBit = magma.Bits(1)
+
+        self.add_ports(
+            pad=magma.InOut(TBit),
+            p2f=magma.Out(TBit),
+            f2p_16=magma.In(TBit),
+            f2p_1=magma.In(TBit),
+            clk=magma.In(magma.Clock),
+            reset=magma.In(magma.AsyncReset),
+            config=magma.In(ConfigurationType(8, 32)),
```

Comment:
If you're planning on instantiating io1bit tiles by themselves at the top level you'll need all 32 bits of config address, along with a tile_id input. Alternatively, you could create io_groups which contain a certain number of io1bits (probably 16), and consider these io_groups to be tiles. The io1bits would be features within these tiles. What's your plan?